src/grappler/ngraph_add_identityn.cc

```diff
@@ -24,37 +24,24 @@ namespace ngraph_bridge {
 Status AddIdentityN(Graph* input_graph, std::set<string> skip_these_nodes) {
   for (auto node : input_graph->op_nodes()) {
     bool fetch_node = false;
-    bool ref_type = false;
     fetch_node = skip_these_nodes.find(node->name()) != skip_these_nodes.end();
     if (fetch_node) {
       NGRAPH_VLOG(5) << "NGTF_OPTIMIZER: Fetch Node " << node->name();
       // Check the number of outputs of the 'fetch_node'
       // Only move further to create an IdentityN node
       // if it is greater than 0
-      // Also, make sure that none of the output types is
-      // a ref type because IdentityN does not support
-      // an input of type ref type
       if (node->num_outputs()) {
         std::vector<NodeBuilder::NodeOut> inputs;
         std::vector<DataType> input_types;
         for (int i = 0; i < node->num_outputs(); i++) {
-          if (IsRefType(node->output_type(i))) {
-            NGRAPH_VLOG(5) << "NGTF_OPTIMIZER: "
-                           << "Datatype for the node output"
-                           << " at index " << i << " is ref type";
-            ref_type = true;
-            break;
-          }
-          input_types.push_back(node->output_type(i));
+          // DT_X_REF = DT_X + 100. kDataTypeRefOffset = 100 here
+          // https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/types.proto
+          input_types.push_back(static_cast<DataType>(
+              node->output_type(i) -
+              (IsRefType(node->output_type(i)) ? kDataTypeRefOffset : 0)));
```

Comment:
if a ref type is detected subtract 100 (kDataTypeRefOffset) from it to get the corresponding type